File: src/main/java/me/viniciusarnhold/altaria/command/Prefixes.java
package me.viniciusarnhold.altaria.command;

import org.jetbrains.annotations.NotNull;

/**
 * Created by Vinicius.
 *
 * @since ${PROJECT_VERSION}
 */
public class Prefixes {

    @NotNull
    private static final String DEFAULT_PREFIX = "!";

    @NotNull
    private static Prefixes ourInstance = new Prefixes();
    @NotNull
    private String currentPrefix = DEFAULT_PREFIX;

    private Prefixes() {

    }

    @NotNull
    public static Prefixes getInstance() {
        return ourInstance;
    }

    @NotNull
    public String current() {
        return currentPrefix;
    }

    @NotNull
    public String changePrefix(@NotNull String newPrefix) {
        return this.currentPrefix = newPrefix;
    }
}

File: commands/src/main/kotlin/commands/Prefixes.kt
package commands

import java.util.*

/**
 * Created by Vinicius.

 * @since ${PROJECT_VERSION}
 */
object Prefixes {
    var currentPrefix = "!"

    fun current(): String {
        return currentPrefix
    }

    fun changePrefix(newPrefix: String): String {
        Objects.requireNonNull(newPrefix)
        this.currentPrefix = newPrefix
        return this.currentPrefix
    }
}
